controllers/flinkcluster/flinkcluster_observer.go

```diff
@@ -501,6 +499,48 @@ func (observer *ClusterStateObserver) observeJobManagerStatefulSet(
 		clusterNamespace, jmStatefulSetName, "JobManager", observedStatefulSet)
 }
 
+func (observer *ClusterStateObserver) observeTaskManager(storageType v1beta1.StorageType, observedClusterState *ObservedClusterState) error {
+	var log = observer.log
+	if storageType == v1beta1.StorageTypePersistent {
+		var observedTmStatefulSet = new(appsv1.StatefulSet)
+		err := observer.observeTaskManagerStatefulSet(observedTmStatefulSet)
+		if err != nil {
+			if client.IgnoreNotFound(err) != nil {
+				log.Error(err, "Failed to get TaskManager StatefulSet")
+				return err
+			}
+			log.Info("Observed TaskManager StatefulSet", "state", "nil")
+			observedTmStatefulSet = nil
+		} else {
+			log.Info("Observed TaskManager StatefulSet", "state", *observedTmStatefulSet)
+			observedClusterState.tmState = TaskManagerState{
+				storageType:   storageType,
+				tmStatefulSet: observedTmStatefulSet,
+				tmDeployment:  nil,
+			}
+		}
+	}
+
+	var observedTmDeployment = new(appsv1.Deployment)
+	err := observer.observeTaskManagerDeployment(observedTmDeployment)
```

Comment:
This should be in the else branch of the `if storageType == v1beta1.StorageTypePersistent` check ?